### DISCUSSION TITLE
Migrate a variant with order history from one product to another

### DISCUSSION THREAD

**Discussion by ivanakcheurov** [saleor/saleor#14195] [Q&A] [answered]
## Problem 1: moving a variant from 1 product to another product.
I have product P1 with variant P1-VX. Later on it turns out that VX is actually more similar to P2-VY and therefore VX needs to be moved from P1 to P2. 
Both P1-VX and P2-VY have been ordered and therefore cannot be simply deleted without losing customer's order history.
How can I do the move through API or UI?

## Problem 2: merging variants of 2 products into 1 product.
Perhaps it is similar to Problem 1 but then if both products have only 1 variant, then moving a variant to another product would cause the "source" product to become empty (no variants) and I suspect Saleor prohibits making products having no variants.
How can I achieve this merge with API or UI?

Thank you.

**Comment by timuric** [comment, marked as answer]
Deleting a variant would not affect order history order lines. However you would not be able to link to the old product page, but you could create a search field that accepts order line sku as parameter and retrieves the current parent product.
